### DISCUSSION TITLE
Invalidate time-sensitive cache entries

### DISCUSSION THREAD

**Discussion by freb** [urql-graphql/urql#2991] [Q&A] [answered]
I have a GraphQL object that stores file meta-data and includes a signed URL where the actual file data can be downloaded. This signed URL has an expiration so at some point the cache entry will become invalid even without a mutation being performed.

Is it possible to invalidate these entries at the end of their validity period from inside the exchange? Or is this something I will need to set a timeout for on any pages making use of these objects?

**Comment by kitten** [comment, marked as answer]
Hiya 👋
I assume you mean in Graphcache, i.e. our normalised cache?

You could try to write a resolver that basically returns that object (or the given ID field for it). Then that resolver can return `undefined` which is basically a signal to tell the cache that the field is uncached/missing which will automatically force a new request to be made

**Reply by freb** [reply]
A little feedback on this.

I tried this approach and confirmed it marks the data as stale. The problem was that requests for updated data don't appear to happen automatically and all the code I tried within a svelte component to refetch the data when stale did not work 100% of the time. It was the least consistent when I used the back and forward mouse buttons to navigate.

I'm not sure if these issues were compounded by the fact that the stale object was a sub-object of the one I was querying.

I ended up working around this issue another way, but it would be nice to know if this is a viable solution if implemented differently.
